xdsl/dialects/mpi.py

```diff
@@ -212,10 +212,10 @@ class Allreduce(MPIBaseOp):
 
     name = "mpi.allreduce"
 
+    send_buffer: Annotated[OptOperand, Attribute]
     recv_buffer: Annotated[Operand, Attribute]
     count: Annotated[Operand, i32]
     datatype: Annotated[Operand, DataType]
-    send_buffer: Annotated[OptOperand, Attribute]
```

Comment:
I must admit, I am surprised that reordering this changes something and I would consider that an xDSL-bug. I think we should investigate this more deeply to understand why the reodering changes something. Could you reduce the testcase and file an issue? Or maybe @math-fehr has a better idea what's happening?

Reply:
Yeah this seems bad.

Reply:
From this [discussion](https://xdsl.zulipchat.com/#narrow/stream/368604-Framework/topic/IRDL.20.7C.20OptOperand.20vs.20Variadics) it seems that this is an op specific problem and not an xDSL related one.